### PR TITLE
Move charter text and update for package requirements

### PR DESCRIPTION
**Pull request by mstackhouse**
Text update for https://github.com/pharmaverse/meetings/issues/28

@rossfarrugia  This updates a bit of the minimum inclusion criteria. There was good text that I moved over for recommendations. Let me know what you think of this.

### CODE REVIEW

File: content/charter/_index.en.md [hunk 1]
```diff
@@ -55,17 +55,7 @@ Contributing to pharmaverse is open to all organizations working within the clin
...
-
-There should be no specific tie to any company-specific statistical computing environment or internal validation/qualification process. The GitHub repos will have CI/CD (e.g. GitHub actions) to run unit tests and test package builds against an open-source container with the pharmaverse endorsed environment layered (e.g. versioned rocker + pharmaverse). The default branch (e.g. master or main) will be protected against merges that fail CI/CD checks. Individual companies are welcomed to layer additional private CI/CD jobs if needed to meet their internal requirements as long as that is done without adding code specific to one company to the public repo. 
-
-Dependency packages to install will be determined by the team and will be clearly documented. When using these, no source code should be copied and other packages shouldn’t be bundled into releases, as we prefer all pharmaverse packages to target permissive licensing (preferably MIT). 
+For more information on development guidance and requirements, see the minimum inclusion criteria and recommended development practices available in https://pharmaverse.org/contribute/lead/
```
Comment (rossfarrugia) [inline content/charter/_index.en.md:58]:
```suggestion
For more information on development guidance and requirements, see the minimum inclusion criteria and recommended practices available via our [Package co-lead guidance](https://pharmaverse.org/contribute/lead/).
```

File: content/contribute/lead/_index.en.md [hunk 2]
```diff
@@ -24,16 +24,17 @@ If you would like to see your package included in pharmaverse, starting in Janua
 
 * __Hosting and Licensing__
   * Packages must have an acceptable open source license, preferably permissive unless a packaged dependency requires a copy-left license is used. The recommended license of pharmaverse at this point in time is the [Apache License Version 2.0](https://www.apache.org/licenses/LICENSE-2.0).
-  * Source code must be hosted publicly, preferably on GitHub. Wherever hosted, there must be a place to publicly report issues.
+  * Source code must be hosted publicly, preferably on GitHub. Wherever hosted, there must be a place to publicly report issues. We ecnourage new package collaborations to use the pharmaverse GitHub organization where possible.
```
Comment (rossfarrugia) [inline content/contribute/lead/_index.en.md:27]:
```suggestion
  * Source code must be hosted publicly, preferably on GitHub. Wherever hosted, there must be a place to publicly report issues. We encourage new package collaborations to use the pharmaverse GitHub organization where possible. 
```

File: content/contribute/lead/_index.en.md [hunk 3]
```diff
@@ -24,16 +24,17 @@ If you would like to see your package included in pharmaverse, starting in Janua
...
 
-* __Version Compatibility__
+* __Version Compatibility and Dependencies__
   * Pharmaverse packages will follow the same R version support schedule as the tidyverse. Tidyverse packages are maintained to generally support the current version, devel version, and four previous minor release versions of R (see post [here](https://www.tidyverse.org/blog/2019/04/r-version-support/))
     * This requirement will _start_ in January 2023, meaning that the minimum version compatibility of R initially will be version 4.2.2 ("Innocent and Trusting"). Packages will not be required to support previous R versions at this point, rather R version 4.2.2 must be supported for another 4 minor releases.
   * Incremental, semantic versioning should be used. We recommend following the strategies detailed in [R Packages](https://r-pkgs.org/lifecycle.html#version)
+  * Dependency packages to install will be determined by the team and will be clearly documented. When using these, bundling of other package's source code should be avoided where possible, as we prefer all pharmaverse packages to target permissive licensing (preferably MIT).
```
Comment (rossfarrugia) [inline content/contribute/lead/_index.en.md:37]:
```suggestion
  * Dependency packages to install will be determined by the team and will be clearly documented. When using these, bundling of other package's source code should be avoided where possible. 
```

File: content/contribute/lead/_index.en.md [hunk 4]
```diff
@@ -56,6 +57,13 @@ If you would like to see your package included in pharmaverse, starting in Janua
...
 adds sufficient value to pharmaverse. 
 
+We additionally have other recommended development practices, that while not required are encourage for new pharmarverse packages or new package collaborations.
+
+* The development team should agree on a similar style of programming for which the pharmaverse site will offer non-binding recommendations (e.g. lintr compatible and with a common vocabulary for pharma-specific standards like CDISC object naming), so as to achieve a consistent look and feel across modules. Note whilst this would be advantageous for new code from the point of launching pharmaverse onwards, it would be unrealistic to expect any existing packages to rework their code base to accommodate this.
+* Where relevant, development efforts would use synthetic CDISC data, and then users from each developer or tester company would take the code modules back to their own restricted systems to test out using real patient data. The learnings from testing would be shared openly, without sharing any patient information.
```
Comment (rossfarrugia) [inline content/contribute/lead/_index.en.md:63]:
```suggestion
* Where relevant, development efforts would use synthetic CDISC data, and then users from each developer or tester company could optionally take the code modules back to their own restricted systems to test out using their data. The learnings from testing would be shared openly, without sharing any patient information.
```

Comment (rossfarrugia) [reply content/contribute/lead/_index.en.md:63]: i don't want this to appear we're encouraging to use patient data given each company has their own rules here with what you can and cannot do with patient data

File: content/charter/_index.en.md [hunk 5]
```diff
@@ -55,17 +55,7 @@ Contributing to pharmaverse is open to all organizations working within the clin
 
 The pharmaverse recommendation is for all our new code collaborations to be hosted under github.com/pharmaverse. This forms a neutral home for all open-source repos falling under our scope. The admins of this org would include pharmaverse council members.
 
-The preferred model of development of new pharmaverse offerings would be to initiate projects as open-source from the start of development or as soon as a stable first draft is available. This is coupled with an expectation of engagement of other companies in ongoing testing and possibly via supporting development efforts with pull requests/issues/feedback. This way we result in more robust code for which many companies have already engaged with during the development process, and it has a higher chance of successful embedding. In addition, the open and transparent development reduces the chance of others creating duplicative parallel packages.
```
Comment (rossfarrugia) [inline content/charter/_index.en.md:58]: i feel like this first paragraph could be left in the charter, as it doesn't seem to repeat with anything on the other page, and i think its general enough to be in a charter

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
nice job @mstackhouse - i left some comments and suggestions for consideration. i'm also going to add an extra commit for the charter page as i spotted some unrelated sections that could be cleaned up whilst we're at it

**Comment by rossfarrugia** [comment]
@mstackhouse there's also a conflict to be resolved before merging

**Comment by rossfarrugia** [comment]
@mstackhouse in our charter we say "Testing of any code has to be a key consideration **and the pharmaverse council will define guidelines around ideal documentation and unit testing strategy to give sufficient reliability**" - do you think we should add anything here or just remove the bold section given we advise to follow https://r-pkgs.org/ anyway. I don't really think we need to be teaching people pkg dev basics

File: content/contribute/lead/_index.en.md (hunk 3 above)
Comment (rossfarrugia) [reply content/contribute/lead/_index.en.md:37]: i also think we could move this line from the charter over to here: `package dependencies should be intentionally limited to the most necessary, such as selected tidyverse packages`

**Comment by mstackhouse** [comment]
> @mstackhouse in our charter we say "Testing of any code has to be a key consideration **and the pharmaverse council will define guidelines around ideal documentation and unit testing strategy to give sufficient reliability**" - do you think we should add anything here or just remove the bold section given we advise to follow https://r-pkgs.org/ anyway? I don't really think we need to be teaching people pkg dev basics

@rossfarrugia  I'd just defer to the R packages book.

File: content/contribute/lead/_index.en.md (hunk 4 above)
Comment (mstackhouse) [reply content/contribute/lead/_index.en.md:63]: I'm fine with that.

**Comment by mstackhouse** [comment]
> @mstackhouse there's also a conflict to be resolved before merging

Resolved

**Comment by rossfarrugia** [comment]
> @rossfarrugia I'd just defer to the R packages book.

Agree @mstackhouse - feel free to delete that bold sentence above from the charter, as it overpromises what we actually provide.

**Review by MichaelRimler — APPROVED** [review]
Looks good to me.

**Comment by mstackhouse** [comment]
> > @rossfarrugia I'd just defer to the R packages book.
> 
> Agree @mstackhouse - feel free to delete that bold sentence above from the charter, as it overpromises what we actually provide.

Updated that too.

**Review by rossfarrugia — APPROVED** [review]

**Pull request merged**